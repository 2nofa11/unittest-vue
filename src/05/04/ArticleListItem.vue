<script setup lang="ts">
export type ItemProps = {
  id: string;
  title: string;
  body: string;
};

defineProps<ItemProps>();
</script>

<template>
  <li data-test="article-list-item">
    <h3 data-test="article-title">{{ title }}</h3>
    <p data-test="article-body">{{ body }}</p>
    <a :href="`/articles/${id}`" data-test="article-link">もっと見る</a>
  </li>
</template>
