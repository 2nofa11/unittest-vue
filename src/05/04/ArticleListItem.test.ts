import { describe, expect, test } from "@jest/globals";
import { mount } from "@vue/test-utils";
import ArticleListItem, { type ItemProps } from "./ArticleListItem.vue";

const item: ItemProps = {
  id: "howto-testing-with-typescript",
  title: "TypeScript を使ったテストの書き方",
  body: "テストを書く時、TypeScript を使うことで、テストの保守性が向上します…",
};

describe("ArticleListItem.vue", () => {
  test("ID に紐づいたリンクが表示される", () => {
    const wrapper = mount(ArticleListItem, { props: item });
    const link = wrapper.find('[data-test="article-link"]');
    expect(link.exists()).toBe(true);
    expect(link.text()).toBe("もっと見る");
    expect(link.attributes("href")).toBe(
      "/articles/howto-testing-with-typescript"
    );
  });

  test("Snapshot: 一覧要素が表示される", () => {
    const wrapper = mount(ArticleListItem, { props: item });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
