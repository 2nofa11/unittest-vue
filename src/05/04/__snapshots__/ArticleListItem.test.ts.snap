// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem.vue Snapshot: 一覧要素が表示される 1`] = `
"<li data-test="article-list-item">
  <h3 data-test="article-title">TypeScript を使ったテストの書き方</h3>
  <p data-test="article-body">テストを書く時、TypeScript を使うことで、テストの保守性が向上します…</p><a href="/articles/howto-testing-with-typescript" data-test="article-link">もっと見る</a>
</li>"
`;
